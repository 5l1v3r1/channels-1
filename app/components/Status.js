// @flow
import React from 'react';
import { IconArrowDropDown } from './core/Icons';
import styles from './Status.scss';

const Status = ({
  title,
  thumbnail,
  toggleChannelList
}: {
  title: string,
  thumbnail: string,
  toggleChannelList: () => void
}) => (
  <div className={styles.status} onClick={toggleChannelList}>
    <img className={styles.thumbnail} src={thumbnail} alt={title} />
    <h1 className={[styles.title, styles['css-truncate'], styles['css-truncate-target']].join(' ')}>{title}</h1>
    <IconArrowDropDown className={styles.iconArrowDropDown} />
  </div>
);

export default Status;
