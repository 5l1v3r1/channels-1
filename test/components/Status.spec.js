import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Status from '../../app/components/Status';

function setup() {
  const props = {
    title: 'DevTips',
    thumbnail: 'https://yt3.ggpht.com/photo.jpg',
    toggleChannelList: jest.fn()
  };

  const enzymeWrapper = shallow(<Status {...props} />);

  return {
    props,
    enzymeWrapper
  };
}

describe('Status', () => {
  it('should render self', () => {
    const { enzymeWrapper } = setup();
    const tree = toJson(enzymeWrapper);

    expect(tree).toMatchSnapshot();
  });

  it('should call toggleChannelList', () => {
    const { props, enzymeWrapper } = setup();

    enzymeWrapper.simulate('click');

    expect(props.toggleChannelList).toBeCalled();
  });
});
