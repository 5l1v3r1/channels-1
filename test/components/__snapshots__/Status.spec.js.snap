// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Status should render self 1`] = `
<div
  className="status"
  onClick={[Function]}
>
  <img
    alt="DevTips"
    className="thumbnail"
    src="https://yt3.ggpht.com/photo.jpg"
  />
  <h1
    className="title css-truncate css-truncate-target"
  >
    DevTips
  </h1>
  <Component
    className="iconArrowDropDown"
  />
</div>
`;
